README.md

```diff
@@ -132,11 +132,13 @@ public interface MyService {
             // Path parameter variable name must match the request path component
             @Request.PathParam UUID myPathParam,
             @Request.PathParam(encoder = MyCustomParamTypeEncoder.class) MyCustomParamType myPathParam2,
-            // converts an custom type into a Multimap<String, String>
+            // converts a custom type into a Multimap<String, String>
             @Request.QueryMap(encoder = MyCustomTypeEncoder.class) MyCustomQueryParamType myCustomQueryParam,
             @Request.Header("Custom-Header") int requestHeaderValue,
             // Headers can be optional
             @Request.Header("Custom-Optional-Header") OptionalInt maybeRequestHeaderValue,
+            // converts a custom type in a Multimap<String, String>
+            @Request.HeaderMap(encoder = MyCustomTypeEncoder.class) MyCustomQueryParamType myCustomHeaderParam,
```

Comment:
I think we're trying to reuse too many names from earlier in the example using `MyCustomQueryParamType` as a header is confusing.

```suggestion
            @Request.HeaderMap(encoder = MyCustomHeaderTypeEncoder.class) MyCustomHeaderType myCustomHeaderParam,
```